torch/_inductor/scheduler.py

```diff
@@ -1722,42 +1770,82 @@ def speedup_by_fusion(self, node1, node2):
 
         why = WhyNoFuse(node1, node2)
 
-        try:
-            ms1, path1 = self.benchmark_fused_nodes(node_list_1)
-            if math.isinf(ms1):
-                why("register spilling of the first kernel")
-                return False
+        def log_fusion(ms_fused, ms1, ms2):
+            if fusion_log.isEnabledFor(logging.DEBUG):
+                if ms_fused < ms1 + ms2:
+                    fusion_log.debug(
+                        "can fuse (benchmark): fusing %s with %s cause %sx speedup",
+                        node1.get_names(),
+                        node2.get_names(),
+                        green_text(f"{(ms1 + ms2) / ms_fused:.3f}"),
+                    )
+                else:
+                    fusion_log.debug(
+                        "cannot fuse (benchmark): fusing %s with %s cause %sx slowdown",
+                        node1.get_names(),
+                        node2.get_names(),
+                        red_text(f"{ms_fused / (ms1 + ms2):.3f}"),
+                    )
+
+        if isinstance(node1, SchedulerNode) and isinstance(
+            node1.node, ir.MultiTemplateBuffer
+        ):
+            multi_node = node1.node
+            choice_timings = multi_node.choice_timings
```

Comment:
hmm, do you need update multi_node.choice_timings with the latency including the fused epilogue?
Let's say we have a sequence: matmul, pointwise1, pointwise2.

Currently code seems to use the latency of matmul rather than (matmul + pointwise1) when we check if  further fusing in  pointwise2 is beneficial

Reply:
We actually only go through this process once, and then we finalize the template. I'll add comments / make clearer